From 25d62725f31ee2f2c9f51d34dda5bcbe467f544e Mon Sep 17 00:00:00 2001
From: Seth Forshee <seth.forshee@canonical.com>
Date: Tue, 23 Aug 2011 13:38:34 -0500
Subject: drm/i915: Remove BUG_ON from i915_gem_evict_something

BugLink: https://bugs.launcpad.net/bugs/828550

According to Chris Wilson this BUG_ON was just paranoia and never
something the kernel should explode on. He later removed it in
upstream commit e39a01501b228e1be2037d5bddccae2a820af902. So let's
just remove it.

Signed-off-by: Seth Forshee <seth.forshee@canonical.com>
Signed-off-by: Stefan Bader <stefan.bader@canonical.com>
---
 drivers/gpu/drm/i915/i915_gem_evict.c |    7 -------
 1 file changed, 7 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_gem_evict.c b/drivers/gpu/drm/i915/i915_gem_evict.c
index 9c1ec78..c4c6b6f 100644
--- a/drivers/gpu/drm/i915/i915_gem_evict.c
+++ b/drivers/gpu/drm/i915/i915_gem_evict.c
@@ -175,13 +175,6 @@ found:
 		list_del_init(&obj_priv->evict_list);
 	}
 
-	/* The just created free hole should be on the top of the free stack
-	 * maintained by drm_mm, so this BUG_ON actually executes in O(1).
-	 * Furthermore all accessed data has just recently been used, so it
-	 * should be really fast, too. */
-	BUG_ON(!drm_mm_search_free(&dev_priv->mm.gtt_space, min_size,
-				   alignment, 0));
-
 	return 0;
 }
 
